### ISSUE TITLE
ioBuffer Memory Leak

### ISSUE THREAD

**Issue by irgnosis** [StackExchange/StackExchange.Redis#748]
We are using StackExchange.Redis v 1.2.3 in our ASP.NET application. The application relies heavily on Azure Redis Cache. After running for a while the application captures ~5GB of memory. The primary reason is ioBuffer byte array. Is this a known issue or we are doing something wrong: Below is our code to interact with Redis
```c#
private static readonly Lazy<ConnectionMultiplexer> lazyConnection = new Lazy<ConnectionMultiplexer>(() => ConnectionMultiplexer.Connect(redisConfiguration));
private static ConnectionMultiplexer Connection => lazyConnection.Value;

IDatabase db= Connection.GetDatabase();
string data= await db.StringGetAsync(key).ConfigureAwait(false);
```

![image](https://user-images.githubusercontent.com/23690556/33644329-e7427ef0-d9f8-11e7-999e-440543293f4d.png)

![image](https://user-images.githubusercontent.com/23690556/33644338-fa971c7c-d9f8-11e7-803e-8c38de7ae9a7.png)

**Comment by mgravell** [comment]
How many connections are we talking about here? I'm confused as to where all those arrays are coming from. Can you clarify?

**Comment by freeusername** [comment]
We're experiencing the same issue, any updates on this?

**Comment by NickCraver** [comment]
Going to close this out as we can't debug this without more info. It seems like the payloads are being held onto, I agree, but we need more info to go on. This may also have been the result of a nasty reader bug from 1.2.1 to 1.2.5, resolved in 1.2.6. If someone is still seeing this on 1.2.6 and above, please ping here and I'll re-open and we'll dig in.

**Comment by PolitovArtyom** [comment]
Have same problem on StackExchange.Redis v 1.2.6 after migration from .Net Framework to .Net Standard 2.0.

Ready to provide necessary info.
